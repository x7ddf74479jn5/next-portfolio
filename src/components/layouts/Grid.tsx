import styles from "src/styles/components/layouts/Grid.module.scss";

export const GridRow: React.FC = ({ children }) => {
  return <div className={styles.gridRow}>{children}</div>;
};
