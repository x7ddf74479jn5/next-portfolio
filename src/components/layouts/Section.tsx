import clsx from "clsx";
import styles from "src/styles/components/layouts/Section.module.scss";

type Props = {
  children: React.ReactNode;
  background?: "primary" | "secondary" | "light";
  id?: string;
  wide?: boolean;
};

export const WrapInSection: React.FC<Props> = ({ children, background, wide = false, ...rest }) => {
  const outerClassName = clsx(styles.section, {
    [styles.bgPrimary]: background === "primary",
    [styles.bgSecondary]: background === "secondary",
    [styles.bgLight]: background === "light",
  });
  const innerClassName = wide ? styles.sectionWrapInWide : styles.sectionWrapIn;
  return (
    <section className={outerClassName} {...rest}>
      <div className={innerClassName}>{children}</div>
    </section>
  );
};
