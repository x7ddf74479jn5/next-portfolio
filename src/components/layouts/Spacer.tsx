import clsx from "clsx";
import styles from "src/styles/components/layouts/Spacer.module.scss";

type Props = {
  size: "xxxs" | "xxs" | "xs" | "sm" | "md" | "lg" | "xl";
};

const Spacer: React.VFC<Props> = ({ size = "md" }) => {
  return (
    <div
      className={clsx({
        [styles.sm]: size === "sm",
        [styles.md]: size === "md",
        [styles.lg]: size === "lg",
      })}
    ></div>
  );
};

export default Spacer;
