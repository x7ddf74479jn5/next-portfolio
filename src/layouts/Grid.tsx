import styles from "src/styles/layouts/Grid.module.scss";

export const GridRow: React.FC = ({ children }) => {
  return <div className={styles.gridRow}>{children}</div>;
};
