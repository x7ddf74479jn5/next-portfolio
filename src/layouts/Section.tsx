import styles from "src/styles/layouts/Section.module.scss";

type Props = {
  children: React.ReactNode;
  id?: string;
};

export const WrapInSection: React.FC<Props> = ({ children, id }) => {
  return (
    <section className={styles.section} id={id}>
      <div className={styles.sectionWrapIn}>{children}</div>
    </section>
  );
};

export const WrapInWideSection: React.FC<Props> = ({ children, id }) => {
  return (
    <section className={styles.section} id={id}>
      <div className={styles.sectionWrapInWide}>{children}</div>
    </section>
  );
};
