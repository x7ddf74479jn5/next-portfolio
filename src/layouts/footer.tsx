// import Image from "next/image";
import Link from "next/link";
import { FooterMenus } from "src/components/common/footer/FooterMenus";
import PandasharkLogo from "src/components/common/PandasharkLogo";
import styles from "src/styles/components/common/PandasharkLogo.module.scss";

export const Footer: React.FC = () => {
  const handle = (event: React.MouseEvent<HTMLHeadingElement, MouseEvent>) => {};

  return (
    <footer className="footer">
      <div className="main c-section">
        <div className="main-inner">
          <section className={styles.logoArea}>
            <PandasharkLogo width={128} height={64} />
            {/* <h1 className={styles.productlogo}>
              <Link href="/">
                <a target="_self">
                  <Image
                    width={128}
                    height={64}
                    alt="パンダシャーク"
                    className="logo-img"
                    src="/img/icons/pandashark_logo_rectangle.png"
                  />
                </a>
              </Link>
            </h1> */}

            {/* <a href="/" target="_self">
                <img alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" />
              </a> */}
            {/* <a className="p-btn-round p-icon__arrow-next" href="/community/" target="_self" role="button">
              Webを学びたい？
            </a> */}
          </section>
          <FooterMenus />

          <section className="foot">
            <ul>
              <Link href="/terms/">
                <a>
                  <li>利用規約</li>
                </a>
              </Link>
              <Link href="/privacy/">
                <a>
                  <li>プライバシーポリシー</li>
                </a>
              </Link>

              {/* <a href="/terms/"><li>利用規約</li></a>
            <a href="/privacy/"><li>プライバシーポリシー</li></a> */}
            </ul>
          </section>
        </div>
      </div>
    </footer>
  );
};
