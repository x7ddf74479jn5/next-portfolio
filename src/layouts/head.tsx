// <!DOCTYPE html>
// <html lang="ja" dir="ltr" locale="ja_ALL" user-region="north-america">
// <head>
//   <meta charset="utf-8">
//   <meta name="viewport" content="initial-scale=1, minimum-scale=1, width=device-width">
//   <title>Pandashark's portfolio</title>
//   <meta name="description" content="Pandasharkのポートフォリオサイトです。">
//   <link href="/css/style.css" rel="stylesheet">
//   <link href="/css/animate.css" rel="stylesheet">
//   <!-- OGP Tag data-->
//   <meta property="og:description" content="Pandasharkのポートフォリオサイトです。">
//   <meta property="og:title" content="Pandashark's portfolio">
//   <meta property="og:image" content="https://pandashark.web.app/assets/img/icons/logo.png">
//   <meta property="og:image:width" content="600">
//   <meta property="og:image:height" content="315">
//   <meta property="og:type" content="website">
//   <meta property="og:url" content="https://pandashark.web.app/">
//   <meta name="twitter:card" content="summary_large_image" />
//   <meta name="twitter:site" content="@pandashark_" />
//   <link rel="shortcut icon" href="/img/icons/favicon.ico">
//   <link rel="apple-touch-icon" sizes="180x180" href="/img/icons/apple-touch-icon.png">
//   <link rel="icon" type="image/png" sizes="32x32" href="/img/icons/favicon-32x32.png">
//   <link rel="icon" type="image/png" sizes="16x16" href="/img/icons/favicon-16x16.png">
//   <link rel="manifest" href="/public/site.webmanifest">
//   <meta name="msapplication-TileColor" content="#da532c">
//   <meta name="theme-color" content="#181a1b">
//   <!-- Carousel slider -->
//   <link href="https://cdnjs.cloudflare.com/ajax/libs/Swiper/4.5.1/css/swiper.min.css" rel="stylesheet">
//   <!-- Font Awesome Icons -->
//   <link href="https://use.fontawesome.com/releases/v5.6.1/css/all.css" rel="stylesheet">
//   <!-- Google Fonts -->
//   <link href="https://fonts.googleapis.com/css?family=Courgette&display=swap&subset=latin-ext" rel="stylesheet">
//   <!-- The core Firebase JS SDK is always required and must be listed first -->
//   <script src="/__/firebase/7.24.0/firebase-app.js"></script>
//   <!-- TODO: Add SDKs for Firebase products that you want to usehttps://firebase.google.com/docs/web/setup#available-libraries -->
//   <script src="/__/firebase/7.24.0/firebase-analytics.js"></script>
//   <script src="/__/firebase/7.24.0/firebase-functions.js"></script>
//   <!-- Initialize Firebase -->
//   <script src="/__/firebase/init.js"></script>
// </head>

import Head from "next/head";

interface Props {
  title: string;
  description: string;
  keyword: string;
  image: string;
  url: string;
  og: {
    type: string;
    image: string;
    imageHight: string;
    imageWidth: string;
    twitterCard: string;
    twitterSite: string;
  };
  icon: string;
  appleTouchIcon: string;
  manifest: string;
}

const headerProps = {
  title: "Pandashark's portfolio",
  description: "Pandasharkのポートフォリオサイトです。",
  keyword: "",
  image: "/img/icons/pandashark_logo.png",
  url: "https://pandashark.web.app/",
  og: {
    type: "website",
    description: "Pandasharkのポートフォリオサイトです。",
    image: "/img/icons/pandashark_logo.png",
    imageHight: "600",
    imageWidth: "315",
    url: "https://pandashark.web.app/",
    twitterCard: "summary_large_image",
    twitterSite: "@pandashark",
  },
  icon: "/img/icons/favicon.ico",
  appleTouchIcon: "/img/icons/apple-touch-icon.png",
  manifest: "",
};

export default ({
  title,
  description,
  keyword,
  image,
  url,
  og,
  icon,
  appleTouchIcon,
  manifest,
}: Props): JSX.Element => {
  return (
    <Head>
      <title>{title}</title>
      <meta property="og:title" content={title} />
      <meta property="og:description" content={description} />
      <meta name="keywords" content={keyword} />
      <meta property="og:type" content="website" />
      <meta property="og:url" content={url} />
      <meta property="og:image" content={image} />
      <meta property="og:image:hight" content={og.imageHight} />
      <meta property="og:image:width" content={og.imageWidth} />
      <meta property="og:site_name" content={title} />
      <meta name="twitter:card" content={og.twitterCard} />
      <meta name="twitter:site" content={og.twitterSite} />
      <meta name="twitter:url" content={url} />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={image} />
      <meta name="msapplication-TileColor" content="#da532c" />
      <meta name="theme-color" content="#181a1b" />
      <link rel="canonical" href={url} />
      {/* <link rel="stylesheet" href={"https://cdnjs.cloudflare.com/ajax/libs/Swiper/4.5.1/css/swiper.min.css"} /> */}
      <link rel="stylesheet" href={"https://use.fontawesome.com/releases/v5.6.1/css/all.css"} />
      <link rel="stylesheet" href={"https://fonts.googleapis.com/css?family=Courgette&display=swap&subset=latin-ext"} />
      <link rel="shortcut icon" href={icon} />
      <link rel="apple-touch-icon" href={appleTouchIcon} />
      <script
        src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
        integrity="sha256-pasqAKBDmFT4eHoN2ndd6lN370kFiGUFyTiUHWhU7k8="
      ></script>
      <script src="/js/animate.js" type="text/javascript"></script>
    </Head>
  );
};
