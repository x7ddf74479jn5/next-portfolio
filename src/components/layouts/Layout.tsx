import * as React from "react";
import { Footer } from "src/components/footer/Footer";
import { Header } from "src/components/header/Header";
import Spacer from "src/components/layouts/Spacer";

export const Layout: React.FC = ({ children }) => {
  return (
    <>
      <Header />
      <Spacer size="md" />
      <main>{children}</main>
      <Footer />
    </>
  );
};
