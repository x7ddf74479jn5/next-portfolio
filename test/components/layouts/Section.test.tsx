import { WrapInSection } from "src/components/layouts/Section";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Section", () => {
  let renderResult: RenderResult;

  describe("primary", () => {
    beforeEach(() => {
      renderResult = render(<WrapInSection background="primary">children</WrapInSection>);
    });
    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.getElementsByTagName("section")[0]).toHaveClass("section bgPrimary");
    });
  });

  describe("secondary", () => {
    beforeEach(() => {
      renderResult = render(<WrapInSection background="secondary">children</WrapInSection>);
    });
    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.getElementsByTagName("section")[0]).toHaveClass("section bgSecondary");
    });
  });

  describe("section bgPrimary", () => {
    beforeEach(() => {
      renderResult = render(<WrapInSection background="light">children</WrapInSection>);
    });
    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.getElementsByTagName("section")[0]).toHaveClass("section bgLight");
    });
  });

  describe("wide and primary", () => {
    beforeEach(() => {
      renderResult = render(
        <WrapInSection wide={true} background="primary">
          children
        </WrapInSection>
      );
    });
    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.getElementsByTagName("div")[0]).toHaveClass("sectionWrapInWide");
    });
  });
});
