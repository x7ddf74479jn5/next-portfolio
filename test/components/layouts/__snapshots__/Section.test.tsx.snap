// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section primary matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgPrimary"
  >
    <div
      class="sectionWrapIn"
    >
      children
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section secondary matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgSecondary"
  >
    <div
      class="sectionWrapIn"
    >
      children
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section section bgPrimary matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgLight"
  >
    <div
      class="sectionWrapIn"
    >
      children
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section wide and primary matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgPrimary"
  >
    <div
      class="sectionWrapInWide"
    >
      children
    </div>
  </section>
</DocumentFragment>
`;
